bindings/apple/MatrixRustSDKTests/MatrixRustSDKTests.swift

```diff
@@ -10,23 +10,14 @@ import XCTest
 
 class MatrixRustSDKTests: XCTestCase {
     
-    static var client: Client!
-    
-    override class func setUp() {
-        client = try! guestClient(basePath: basePath, homeserver: "https://matrix.org")
-    }
-    
-    func testClientProperties() {
-        XCTAssertTrue(Self.client.isGuest())
-        
-        XCTAssertNotNil(try? Self.client.restoreToken())
-        XCTAssertNotNil(try? Self.client.deviceId())
-        XCTAssertNotNil(try? Self.client.displayName())
-    }
-    
     func testReadOnlyFileSystemError() {
         do {
-            let _ = try loginNewClient(basePath: "", username: "test", password: "test")
+            let client = try ClientBuilder()
+                .basePath(basePath: "")
+                .username(username: "@test:domain")
+                .build()
+            
+            try client.login(username: "@test:domain", password: "test")
```

Comment:
The use of the new ClientBuilder API highlights that username now has to be passed twice, which is not ideal. The client builder needs username to determine file store path, and potentially homeserver, if not set explicitely. The client needs username to construct user to log in with.

Potentially the client could retrieve the username directly from the builder, but not sure if api like `client.login(password)` is very good.